### PR TITLE
Workflow test run

### PR DESCRIPTION
**Pull request by dimittal**

### CODE REVIEW

File: executor/workflows/tasks.py [hunk 1]
```diff
@@ -254,16 +259,66 @@ def test_workflow_notification(account_id, workflow, message_type):
...
-        execution_output: [InterpretationProto] = playbook_execution_result_interpret(InterpreterType.BASIC_I, p_proto,
-                                                                                      pe_logs)
+        execution_output: [InterpretationProto] = playbook_execution_result_interpret(InterpreterType.BASIC_I, p_proto, pe_logs)
+        action_executor(account, workflow.actions[0], execution_output)
+
+    if message_type == WorkflowActionSlackNotificationConfig.MessageType.THREAD_REPLY:
+
+        # Sample alert message
+        channel_id = workflow.entry_points[0].alert_config.slack_channel_alert_config.slack_channel_id.value
+        bot_auth_token = ConnectorKey.objects.get(key_type=ConnectorKeyProto.KeyType.SLACK_BOT_AUTH_TOKEN, is_active=True).key
```
Comment (droid-mohit) [inline executor/workflows/tasks.py:279]: use the helper functions in crud files, as it handles exception failures. Currently this code will throw an error if the key is not found and its not handled.

File: executor/workflows/tasks.py [hunk 2]
```diff
@@ -254,16 +259,66 @@ def test_workflow_notification(account_id, workflow, message_type):
...
+        message_ts = SlackApiProcessor(bot_auth_token).send_bot_message(channel_id, 'Hello, this is a test alert message from to show how the enrichment works in reply to an alert.')
```
Comment (droid-mohit) [inline executor/workflows/tasks.py:280]: Hello, this is a test alert message from (insert - MyDroid or Doctor Droid here) to show how the enrichment works in reply to an alert.

File: executor/workflows/tasks.py [hunk 3]
```diff
@@ -254,16 +259,66 @@ def test_workflow_notification(account_id, workflow, message_type):
...
+
+        workflow.actions[0].notification_config.slack_config.thread_ts.value = message_ts
+
+        pe_logs = []
+        account = Account.objects.get(id=account_id)
+        time_range = { "time_lt": str(int(time.time())), "time_geq": str(int(time.time()) - 3600) }
```
Comment (droid-mohit) [inline executor/workflows/tasks.py:286]: Why not create the proto directly here?

File: executor/workflows/tasks.py [hunk 4]
```diff
@@ -254,16 +259,66 @@ def test_workflow_notification(account_id, workflow, message_type):
             for step, all_task_results in all_step_executions.items():
                 for result in all_task_results:
                     playbook_execution_log = PlaybookExecutionLog(
-                        playbook=playbook,
-                        playbook_step=step,
-                        playbook_task_definition=result['task'].proto,
-                        playbook_task_result=result['task_result_proto'],
+                        playbook=p_proto,
+                        step=step.proto,
+                        task=result['task'].proto,
```
Comment (droid-mohit) [inline executor/workflows/tasks.py:264]: On line 254 we are deliberately converting the proto to dict while we are doing the reverse here. Change the code on line 254 to not convert the task into a dict

File: executor/workflows/tasks.py [hunk 5]
```diff
@@ -254,16 +259,66 @@ def test_workflow_notification(account_id, workflow, message_type):
...
                     )
                     pe_logs.append(playbook_execution_log)
 
         except Exception as exc:
             logger.error(f"Error occurred while running playbook: {exc}")
 
-        execution_output: [InterpretationProto] = playbook_execution_result_interpret(InterpreterType.BASIC_I, p_proto,
-                                                                                      pe_logs)
+        execution_output: [InterpretationProto] = playbook_execution_result_interpret(InterpreterType.BASIC_I, p_proto, pe_logs)
+        action_executor(account, workflow.actions[0], execution_output)
```
Comment (droid-mohit) [inline executor/workflows/tasks.py:273]: In all the places where we are accessing a list, add an empty list check to protect from unwanted exceptions breaking the code.

File: executor/workflows/tasks.py (hunk 4 above)
Comment (droid-mohit) [reply executor/workflows/tasks.py:264]: On line 247 do we care about the ordering?

File: executor/workflows/tasks.py (hunk 5 above)
Comment (dimittal) [reply executor/workflows/tasks.py:273]: i have a lot of these array validations in the view before calling this function

File: executor/workflows/tasks.py (hunk 4 above)
Comment (dimittal) [reply executor/workflows/tasks.py:264]: Removed ordering, removed redundant proto <-> dict conversions

File: executor/workflows/tasks.py (hunk 3 above)
Comment (dimittal) [reply executor/workflows/tasks.py:286]: Changed

File: executor/workflows/tasks.py (hunk 2 above)
Comment (dimittal) [reply executor/workflows/tasks.py:280]: Can't find a way to self tag. Fixed the grammar

File: executor/workflows/tasks.py (hunk 1 above)
Comment (dimittal) [reply executor/workflows/tasks.py:279]: Done

**Pull request merged**